### PR TITLE
parser: Fix parsing one-line enum declarations 

### PR DESCRIPTION
**Pull request by MacDue**

### CODE REVIEW

File: tests/parser/one-line-enum-no-comma-ending.jakt [hunk 1]
```diff
@@ -0,0 +1 @@
+enum Fruit { Apple }
```
Comment (jntrnr) [inline tests/parser/one-line-enum-no-comma-ending.jakt:1]: this is valid Jakt - are you thinking we should make it invalid?

Comment (MacDue) [reply tests/parser/one-line-enum-no-comma-ending.jakt:1]: It was an assumption based off the error that seemed to indicate that variants should end with a comma or newline.
I'll correct that.

Comment (MacDue) [reply tests/parser/one-line-enum-no-comma-ending.jakt:1]: (Currently enums like `enum Fruit { Apple }` don't parse correctly, so I was not 100% sure on the correct parse)

**Comment by MacDue** [comment]
I've changed the PR to now allow enum declarations like `enum Fruit { Apple }`, and parse them correctly.  

I've also added a new sample that failed to compile before these changes.

**Comment by jntrnr** [comment]
thanks for the fix!

**Pull request merged**